src/gtest/test_merkletree.cpp

```diff
@@ -322,3 +323,36 @@ TEST(orchardMerkleTree, appendBundle) {
         EXPECT_EQ(newTree.root(), readBack.root());
     }
 }
+
+TEST(saplingMerkleTree, subtree_roots)
+{
+    SaplingMerkleTree tree;
+    for (size_t i = 0; i < (1 << 16); i++) {
+        EXPECT_FALSE(tree.complete_subtree_root().has_value());
+        EXPECT_EQ(tree.current_subtree_index(), 0);
+        AppendRandomLeaf(tree);
+    }
+    EXPECT_EQ(tree.current_subtree_index(), 1);
+    auto subtree_root = tree.complete_subtree_root();
+    ASSERT_TRUE(subtree_root.has_value());
+    uint256 cur = *subtree_root;
+    for (size_t depth = 16; depth < 32; depth++) {
```

Comment:
Non-blocking nit: the `16`s in this test should be derived from `libzcash::TRACKED_SUBTREE_HEIGHT` (either directly, or as a subtraction from `SAPLING_INCREMENTAL_MERKLE_TREE_DEPTH`, as appropriate). Likewise, the `32` should be `SAPLING_INCREMENTAL_MERKLE_TREE_DEPTH`.